x-pack/legacy/plugins/code/server/distributed/code_services.test.ts

```diff
@@ -80,13 +82,13 @@ test('non-code-node could send request to code-node', async () => {
     baseUrl: string,
     path: string,
     payload: RequestPayload,
-    originRequest: Request
+    originRequest: KibanaRequest
   ) => {
     expect(baseUrl).toBe(codeNodeUrl);
     const response = await hapiServer.inject({
```

Comment:
@restrry we will need the mock for this one to replace the hapiServer here.

Reply:
You already mocked router in the test. If you want to add an integration test instead, you can use `kbnTestServer`. There is an example how to do it https://github.com/elastic/kibana/blob/master/src/core/server/legacy/integration_tests/legacy_service.test.ts